src/server/qtquick/wsurfaceitem.cpp

```diff
@@ -880,10 +882,9 @@ void WSurfaceItem::releaseResources()
     if (!d->surfaceFlags.testFlag(DontCacheLastBuffer)) {
         for (auto item : d->subsurfaces) {
             item->releaseResources();
-            // Don't auto destroy subsurfaes's items, ensure save the
             // subsurface's contents at the last frame buffer.
-            QObject::disconnect(item->surface(), &QWSurface::destroyed,
-                                item, &WSurfaceItem::deleteLater);
+            // AutoDestroy: disconnects (subsurfaceItem.surface, destroyed, this, lambda{deleteLater})
+            Q_ASSERT( item->surface()->disconnect(this) );
```

Comment:
`Q_ASSERT` will ignore the code on release mode, so you can't call function in `Q_ASSERT`.

Reply:
I think use `QObject::disconnect(item->surface(), &WSurface::destroy, this, nullptr);` is better here, Disconnect all connections is dangerous, because maybe other developer will needs connect other signals in future and he don't wants disconnect it at here, but he unknown exists disconnect behavior here.